ci-operator/step-registry/cucushift/installer/rehearse/aws/cases/multi-clusters-one-phz/provision/cucushift-installer-rehearse-aws-cases-multi-clusters-one-phz-provision-commands.sh

```diff
@@ -0,0 +1,527 @@
+
+trap 'save_artifacts' EXIT TERM INT
+
+function save_artifacts()
+{
+  set +o errexit
+  current_time=$(date +%s)
+  cp "${install_dir1}/metadata.json" "${SHARED_DIR}/cluster-1-metadata.json"
+  cp "${install_dir2}/metadata.json" "${SHARED_DIR}/cluster-2-metadata.json"
+
+  cp "${install_dir1}"/log-bundle-*.tar.gz "${ARTIFACT_DIR}/" 2>/dev/null
+  cp "${install_dir2}"/log-bundle-*.tar.gz "${ARTIFACT_DIR}/" 2>/dev/null
+
+  sed '
+    s/password: .*/password: REDACTED/;
+    s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+    s/UserData:.*,/UserData: REDACTED,/;
+    ' "${install_dir1}/.openshift_install.log" > "${ARTIFACT_DIR}/cluster_1_openshift_install-${current_time}.log"
+  
+  sed '
+    s/password: .*/password: REDACTED/;
+    s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+    s/UserData:.*,/UserData: REDACTED,/;
+    ' "${install_dir2}/.openshift_install.log" > "${ARTIFACT_DIR}/cluster_2_openshift_install-${current_time}.log"
+
+  
+  if [ -d "${install_dir1}/.clusterapi_output" ]; then
+    mkdir -p "${ARTIFACT_DIR}/cluster_1_clusterapi_output-${current_time}"
+    cp -rpv "${install_dir1}/.clusterapi_output/"{,**/}*.{log,yaml} "${ARTIFACT_DIR}/cluster_1_clusterapi_output-${current_time}" 2>/dev/null
+  fi
+
+  if [ -d "${install_dir2}/.clusterapi_output" ]; then
+    mkdir -p "${ARTIFACT_DIR}/cluster_2_clusterapi_output-${current_time}"
+    cp -rpv "${install_dir2}/.clusterapi_output/"{,**/}*.{log,yaml} "${ARTIFACT_DIR}/cluster_2_clusterapi_output-${current_time}" 2>/dev/null
+  fi
+
+  set -o errexit
+}
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+REGION=${LEASED_RESOURCE}
+CLUSTER_PREFIX="${NAMESPACE}-${UNIQUE_HASH}"
+ROUTE53_HOSTED_ZONE_NAME="${CLUSTER_PREFIX}.${BASE_DOMAIN}"
+VPC_ID=$(cat "${SHARED_DIR}/vpc_id")
+subnet_ids_file="${SHARED_DIR}/subnet_ids"
+# -----------------------------------------
+# Creating PHZ
+# -----------------------------------------
+
+
+# Use a timestamp to ensure the caller reference is unique, as we've found
+# cluster name can get reused in specific situations.
+TIMESTAMP=$(date +%s)
+CALLER_REFERENCE_STR="${ROUTE53_HOSTED_ZONE_NAME}-${TIMESTAMP}"
+
+echo -e "creating route53 hosted zone: ${ROUTE53_HOSTED_ZONE_NAME}"
+HOSTED_ZONE_CREATION=$(aws --region "$REGION" route53 create-hosted-zone --name "${ROUTE53_HOSTED_ZONE_NAME}" --vpc VPCRegion="${REGION}",VPCId="${VPC_ID}" --caller-reference "${CALLER_REFERENCE_STR}")
+
+HOSTED_ZONE_ID="$(echo "${HOSTED_ZONE_CREATION}" | jq -r '.HostedZone.Id' | awk -F / '{printf $3}')"
+# save hosted zone information to ${SHARED_DIR} for deprovision step
+echo "${HOSTED_ZONE_ID}" > "${SHARED_DIR}/hosted_zone_id"
+CHANGE_ID="$(echo "${HOSTED_ZONE_CREATION}" | jq -r '.ChangeInfo.Id' | awk -F / '{printf $3}')"
+
+# add a sleep time to reduce Rate exceeded errors
+sleep 120
+
+aws --region "${REGION}" route53 wait resource-record-sets-changed --id "${CHANGE_ID}" &
+wait "$!"
+echo "Hosted zone ${HOSTED_ZONE_ID} successfully created."
```

Comment:
Can we re-use `aws-provision-route53-private-hosted-zone` step instead of the above steps of `Creating PHZ`

Reply:
Updated